backend/endpoints/rooms.py

```diff
@@ -219,17 +220,19 @@ async def join_room(
     if not room or room["room_type"].upper() == RoomType.DM:
         raise HTTPException(
             status_code=status.HTTP_403_FORBIDDEN,
-            detail="DM room cannot be joined or not found",
+            detail="room not found" if not room else "DM room cannot be joined",
         )
 
     member = room.get("room_members").get(str(member_id))
-    if member is None or member["role"].lower() != Role.ADMIN:
-        raise HTTPException(
-            status_code=status.HTTP_401_UNAUTHORIZED,
-            detail="member not in room or not an admin",
-        )
 
     if room["room_type"].upper() == RoomType.CHANNEL:
+        if room["is_private"] is True and (
+            member is None or member["role"].lower() != Role.ADMIN
+        ):
+            raise HTTPException(
+                status_code=status.HTTP_401_UNAUTHORIZED,
+                detail="only admins can add new members",
+            )
         room["room_members"].update(members)
 
     if room["room_type"].upper() == RoomType.GROUP_DM:
```

Comment:
I've tested the endpoint with these values:
# Parameters
```
org_id == 619ba4671a5f54782939d384
room_id == 61ca29c478fb01b18fac1477 (public channel, with room_members == None)
member_id == 619e28c31a5f54782939d59a (not a member in the room_id's members)
```

# Request body
```
{
  "additionalProp1": {
    "role": "member",
    "starred": false,
    "closed": false
  },
  "additionalProp2": {
    "role": "member",
    "starred": false,
    "closed": false
  },
  "additionalProp3": {
    "role": "member",
    "starred": false,
    "closed": false
  }
}
```

But the endpoint ran successfully with these values.
We should add some additional checks.
What do you think about it? I also suggest another PR to fix this issue.

Reply:
wow @zxenonx that's so true o!

Reply:
an additional check will complicate this endpoint further sha, because the only way I can think of right now is checking if the member is the organization, which will require fetching all members in the org via a call to zc_core, smh.
do you have a different suggestion? @zxenonx

Reply:
but that check is very necessary

Reply:
Yeah @mykie88 , the check can be done in a separate function and called in the endpoint. 
But for now, I don't see any other solutions than what you suggested.
We can put our heads together and come up with the best way to solve this.

Reply:
Yeah that will be great. Thanks @zxenonx

Reply:
@o4codes @pauline-banye @man-of-mind @kinglighthill @AI-fae